spark/src/test/scala/org/apache/spark/sql/delta/clustering/ClusteringTableFeatureSuite.scala

```diff
@@ -49,24 +60,60 @@ class ClusteringTableFeatureSuite extends SparkFunSuite with DeltaSQLCommandTest
     }
   }
 
-  test("alter table cluster by non-clustered tables is not allowed.") {
+  test("alter table cluster by partitioned tables is not allowed.") {
     withTable("tbl") {
-      sql("CREATE TABLE tbl(a INT, b STRING) USING DELTA")
+      sql("CREATE TABLE tbl(a INT, b STRING) USING DELTA PARTITIONED BY (a)")
       val e1 = intercept[DeltaAnalysisException] {
         sql("ALTER TABLE tbl CLUSTER BY (a)")
       }
       checkError(
         e1,
-        "DELTA_ALTER_TABLE_CLUSTER_BY_NOT_ALLOWED",
+        "DELTA_ALTER_TABLE_CLUSTER_BY_ON_PARTITIONED_TABLE_NOT_ALLOWED",
         parameters = Map.empty)
 
       val e2 = intercept[DeltaAnalysisException] {
         sql("ALTER TABLE tbl CLUSTER BY NONE")
       }
       checkError(
         e2,
-        "DELTA_ALTER_TABLE_CLUSTER_BY_NOT_ALLOWED",
+        "DELTA_ALTER_TABLE_CLUSTER_BY_ON_PARTITIONED_TABLE_NOT_ALLOWED",
         parameters = Map.empty)
     }
   }
+
+   test("alter table cluster by unpartitioned non-clustered tables is supported.") {
```

Comment:
I find `unpartitioned non-clustered` a bit convoluted, shall we just say `unpartitioned ` tables? I think we have the percention of delta tables as :
1. Partitioned table
2. Unpartitioned table
3. Clustered table